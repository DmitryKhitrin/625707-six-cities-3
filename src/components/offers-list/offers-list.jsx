import React, {memo} from "react";
import {PropTypes} from "prop-types";
import {useHistory} from "react-router-dom";
import {PlaceCard} from "../place-card/place-card";

const OffersList = ({
  placeCardsList,
  onMouseEnter = () => {},
  onMouseLeave = () => {},
  setFavorite,
  isAuthenticated,
}) => {
  const history = useHistory();
  return (
    <div className="cities__places-list places__list tabs__content">
      {placeCardsList.map(
          ({
            id,
            price,
            previewImage,
            title,
            rating,
            type,
            isPremium,
            isFavorite
          }) => {
            const onFavClick = () =>
              isAuthenticated ? setFavorite(id, Number(!isFavorite)) : history.push(`/login`);
            return (
              <div key={id}>
                <PlaceCard
                  id={id}
                  isPremium={isPremium}
                  type={type}
                  rating={rating}
                  price={price}
                  previewImage={previewImage}
                  title={title}
                  onMouseEnter={onMouseEnter}
                  onMouseLeave={onMouseLeave}
                  isFavorite={isFavorite}
                  setFavorite={onFavClick}
                />
              </div>
            );
          }
      )}
    </div>
  );
};

const MemoizedOffersList = memo(OffersList);
export {MemoizedOffersList as OffersList};

OffersList.propTypes = {
  placeCardsList: PropTypes.array.isRequired,
  onMouseEnter: PropTypes.func.isRequired,
  onMouseLeave: PropTypes.func.isRequired,
  setFavorite: PropTypes.func,
  isAuthenticated: PropTypes.bool
};
