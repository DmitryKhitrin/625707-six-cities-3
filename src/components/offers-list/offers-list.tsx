import React, {memo, FC} from "react";
import {useHistory} from "react-router-dom";
import {PlaceCard} from "../place-card/place-card";
import {ParsedOfferCard} from "../../utils/utils";

type Props = {
  placeCardsList: ParsedOfferCard[];
  onMouseEnter: (T: string) => void;
  onMouseLeave: () => void;
  setFavorite: (id: string, hotel: number) => void;
  isAuthenticated: boolean;
}

const OffersList: FC<Props> = ({
  placeCardsList,
  onMouseEnter = () => {},
  onMouseLeave = () => {},
  setFavorite,
  isAuthenticated,
}) => {
  const history = useHistory();
  return (
    <div className="cities__places-list places__list tabs__content">
      {placeCardsList.map(
          ({
            id,
            price,
            previewImage,
            title,
            rating,
            type,
            isPremium,
            isFavorite
          }) => {
            const handleFavoriteClick = () =>
              isAuthenticated ? setFavorite(id, Number(!isFavorite)) : history.push(`/login`);
            return (
              <div key={id}>
                <PlaceCard
                  id={id}
                  isPremium={isPremium}
                  type={type}
                  rating={rating}
                  price={price}
                  previewImage={previewImage}
                  title={title}
                  onMouseEnter={onMouseEnter}
                  onMouseLeave={onMouseLeave}
                  isFavorite={isFavorite}
                  onFavoriteClick={handleFavoriteClick}
                />
              </div>
            );
          }
      )}
    </div>
  );
};

const MemoizedOffersList = memo(OffersList);
export {MemoizedOffersList as OffersList};
