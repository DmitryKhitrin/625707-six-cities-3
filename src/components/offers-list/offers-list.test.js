import React from "react";
import renderer from "react-test-renderer";
import {OffersList} from "./offers-list.jsx";

const mocData = [
  {
    id: `1`,
    priceValue: 1,
    placeCardImage: `img/apartment-01.jpg`,
    starsRating: `25%`,
    roomType: `Apartment`,
    isPremium: true,
    cardName: `Just Flat.`
  },
  {
    id: `2`,
    priceValue: 20,
    placeCardImage: `img/room.jpg`,
    starsRating: `50%`,
    roomType: `Just Flat`,
    isPremium: false,
    cardName: `Just plane text.`
  },
  {
    id: `3`,
    priceValue: 100,
    placeCardImage: `img/apartment-02.jpg`,
    starsRating: `100%`,
    roomType: `Private room`,
    isPremium: true,
    cardName: `Best place in the World really.Beautiful luxurious apartment at great location`
  }
];

const onHeaderClick = () => {};

it(`<OffersList /> should render OffersList.`, () => {
  const tree = renderer
    .create(
        <OffersList
          placeCardsList={mocData}
          onHeaderClick={onHeaderClick}
          onMouseEnter={jest.fn()}
          onMouseLeave={jest.fn()}
        />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
