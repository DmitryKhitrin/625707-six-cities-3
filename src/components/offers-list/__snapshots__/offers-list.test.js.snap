// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OffersList /> should render OffersList. 1`] = `
<div
  className="cities__places-list places__list tabs__content"
>
  <div>
    <article
      className="cities__place-card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        className="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="/offer/1"
          onClick={[Function]}
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src="img/apartment-01.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              1
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button button"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "25%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/offer/1"
            onClick={[Function]}
          >
            Just Flat.
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          Apartment
        </p>
      </div>
    </article>
  </div>
  <div>
    <article
      className="cities__place-card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="/offer/2"
          onClick={[Function]}
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src="img/room.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              20
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button button"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "50%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/offer/2"
            onClick={[Function]}
          >
            Just plane text.
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          Just Flat
        </p>
      </div>
    </article>
  </div>
  <div>
    <article
      className="cities__place-card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        className="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="/offer/3"
          onClick={[Function]}
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src="img/apartment-02.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              100
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button button"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/offer/3"
            onClick={[Function]}
          >
            Best place in the World really.Beautiful luxurious apartment at great location
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          Private room
        </p>
      </div>
    </article>
  </div>
</div>
`;
